File: Apps/Android/monitor-ui/src/main/kotlin/com/opencapture/monitorui/MonitorAssistIcon.kt
package com.opencapture.monitorui

import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource

/** Exact paths extracted from the approved Field Monitor HTML, never approximations. */
enum class MonitorAssistIcon(val drawable: Int) {
    LUT(R.drawable.monitor_assist_lut), PEAKING(R.drawable.monitor_assist_peaking),
    FALSE_COLOR(R.drawable.monitor_assist_false_color), ZEBRA(R.drawable.monitor_assist_zebra),
    WAVEFORM(R.drawable.monitor_assist_waveform), RGB_PARADE(R.drawable.monitor_assist_rgb_parade),
    HISTOGRAM(R.drawable.monitor_assist_histogram), VECTORSCOPE(R.drawable.monitor_assist_vectorscope),
    TRAFFIC_LIGHTS(R.drawable.monitor_assist_traffic_lights), FRAME_GUIDE(R.drawable.monitor_assist_frame_guide),
    GRID(R.drawable.monitor_assist_grid), CROSSHAIR(R.drawable.monitor_assist_crosshair),
    MIRROR(R.drawable.monitor_assist_mirror), AUDIO_METERS(R.drawable.monitor_assist_audio_meters),
}

@Composable
fun MonitorAssistIcon(icon: MonitorAssistIcon, tint: Color, modifier: Modifier = Modifier,
    contentDescription: String? = null) {
    Icon(painterResource(icon.drawable), contentDescription, modifier, tint)
}

File: Sources/MonitorUI/MonitorAssistIcon.swift
#if os(iOS)
    import SwiftUI

    /// Exact view-assist artwork from the approved Field Monitor prototype. These
    /// deliberately form a separate catalog from the surrounding Lucide UI.
    public enum MonitorAssistIcon: String, CaseIterable, View {
        case lut
        case peaking
        case falseColor = "false-color"
        case zebra
        case waveform
        case rgbParade = "rgb-parade"
        case histogram
        case vectorscope
        case trafficLights = "traffic-lights"
        case frameGuide = "frame-guide"
        case grid
        case crosshair
        case mirror
        case audioMeters = "audio-meters"

        public var body: some View {
            Canvas { context, size in
                Self.documents[self]?.draw(in: &context, size: size, filled: false)
            }
            .aspectRatio(1, contentMode: .fit)
            .accessibilityHidden(true)
        }

        // All 14 authoritative glyphs are single paths, fill="none", opacity 1,
        // stroke="currentColor", stroke-width="2", round caps and joins. Loading
        // the SVG retains the prototype's exact geometry and stroke treatment.
        // Parsing occurs once per catalog, never during an animation or feed tick.
        @MainActor static let documents: [Self: LucideSVGDocument] = Dictionary(
            uniqueKeysWithValues: allCases.compactMap { icon in
                let stem = "monitor-assist-" + icon.rawValue
                guard
                    let url = Bundle.module.url(
                        forResource: stem, withExtension: "svg", subdirectory: "Icons/assist"),
                    let xml = try? String(contentsOf: url, encoding: .utf8),
                    let document = LucideSVGDocument(xml: xml)
                else { return nil }
                return (icon, document)
            })
    }
#endif
